### PR TITLE
CAAM support with QNX i.MX8, add AES-CTR crypto callback

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
# Description

Adds support for i.MX8 + QNX + CAAM. Adds AES GCM/CCM/CBC/ECB modes to caam_driver.c.

# Testing

i.MX8 running wolfcrypt/test/test.c

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
retest this Jenkins

File: wolfcrypt/src/port/caam/caam_driver.c [hunk 1]
```diff
@@ -285,14 +302,14 @@ int caamFindUnusedPartition()
 /* flag contains how the parition is set i.e CSP flag and read/write access
  *      it also contains if locked
  */
-static Error caamCreatePartition(unsigned int page, unsigned int par,
+static Error caamCreatePartition(unsigned int* page, unsigned int par,
```
Comment (dgarske) [inline wolfcrypt/src/port/caam/caam_driver.c:305]: Above spelling. `parition` -> `partition`

File: wolfcrypt/src/port/caam/caam_driver.c [hunk 2]
```diff
@@ -342,7 +368,7 @@ CAAM_ADDRESS caamGetPartition(unsigned int part, int partSz, unsigned int flag)
     (void)flag; /* flag is for future changes to flag passed when creating */
 
     /* create and claim the partition */
-    err = caamCreatePartition(part, part, CAAM_SM_CSP | CAAM_SM_SMAP_LOCK |
+    err = caamCreatePartition(&part, part, CAAM_SM_CSP | CAAM_SM_SMAP_LOCK |
```
Comment (dgarske) [inline wolfcrypt/src/port/caam/caam_driver.c:371]: Below spelling `partiions` -> `partitions`

File: wolfcrypt/src/port/caam/caam_driver.c [hunk 3]
```diff
@@ -494,14 +515,14 @@ int caamInitRng(struct CAAM_DEVICE* dev)
...
+        reg = CAAM_READ(caam.baseAddr + CAAM_RTMCTL) & (~CAAM_PRGM);
+        CAAM_WRITE(caam.baseAddr + CAAM_RTMCTL, reg);
+        reg = CAAM_READ(caam.baseAddr + CAAM_RTMCTL);
         reg |= CAAM_CTLERR;
-        CAAM_WRITE(CAAM_RTMCTL, reg);
+        CAAM_WRITE(caam.baseAddr + CAAM_RTMCTL, reg);
 
         /* check out the status and see if already setup */
-        reg = CAAM_READ(CAAM_RDSTA);
+        reg = CAAM_READ(caam.baseAddr + CAAM_RDSTA);
```
Comment (dgarske) [inline wolfcrypt/src/port/caam/caam_driver.c:525]: Below `determinstic` -> `deterministic`

File: wolfcrypt/src/ecc.c [hunk 4]
```diff
@@ -9619,22 +9620,31 @@ int wc_ecc_export_ex(ecc_key* key, byte* qx, word32* qxLen,
...
-            if (key->blackKey > 0) {
+        if (key->blackKey == CAAM_BLACK_KEY_CCM) {
                 if (*dLen < keySz + WC_CAAM_MAC_SZ) {
                     *dLen = keySz + WC_CAAM_MAC_SZ;
                     return BUFFER_E;
                 }
 
-                if (caamReadPartition(key->blackKey, d, keySz + WC_CAAM_MAC_SZ) != 0)
-                    return WC_HW_E;
+                    err = wc_export_int(&key->k, d, dLen, keySz + WC_CAAM_MAC_SZ, encType);
```
Comment (dgarske) [inline wolfcrypt/src/ecc.c:9629]: Alignment off

File: wolfcrypt/src/ecc.c [hunk 5]
```diff
@@ -9619,22 +9620,31 @@ int wc_ecc_export_ex(ecc_key* key, byte* qx, word32* qxLen,
...
+                if (*dLen < keySz + WC_CAAM_MAC_SZ) {
+                    *dLen = keySz + WC_CAAM_MAC_SZ;
+                    return BUFFER_E;
+                }
+
+                if (key->blackKey != CAAM_BLACK_KEY_CCM) {
+                    if (caamReadPartition(key->blackKey, d, keySz + WC_CAAM_MAC_SZ) != 0)
+                        return WC_HW_E;
+                }
+                else {
```
Comment (dgarske) [inline wolfcrypt/src/ecc.c:9644]: Should this blank else {} be removed?

**Review by dgarske — CHANGES_REQUESTED** [review]
Looks great. Couple minor cleanups and should be ready to merge.

**Review by dgarske — APPROVED** [review]

**Pull request merged**